### PR TITLE
Fix issues with auth

### PR DESCRIPTION
**Pull request by Flanker32**
- Close tool window listener when close project
- Don't throw warning again when login with account with no subscription

### CODE REVIEW

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java [hunk 1]
```diff
@@ -116,10 +119,14 @@ private static String composeName() {
                         : selectedSubscriptions.get(0).getSubscriptionName());
             }
         } catch (Exception e) {
-            String msg = "An error occurred while getting the subscription list." + "\n" + "(Message from Azure:" + e
-                    .getMessage() + ")";
-            DefaultLoader.getUIHelper().showException(msg, e,
-                    "MS Services - Error Getting Subscriptions", false, true);
+            if (e instanceof AuthException && e.getMessage().equalsIgnoreCase("No subscription found in the account")) {
```
Comment (andxu) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java]: use stringutil to compare string

File: Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java [hunk 2]
```diff
@@ -86,12 +86,24 @@ public void addSignInEventListener(Runnable l) {
...
+    }
+
     public void addSignOutEventListener(Runnable l) {
         if (!signOutEventListeners.contains(l)) {
             signOutEventListeners.add(l);
         }
     }
 
+    public void removeSignOutEventListener(Runnable l) {
+        if (!signOutEventListeners.contains(l)) {
```
Comment (andxu) [inline Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java]: the logic is wrong

Comment (Flanker32) [reply Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java]: Thanks, fixed~

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java (hunk 1 above)
Comment (Flanker32) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java]: Done

**Pull request merged**